src/PrestaShopBundle/Form/Admin/Type/TextWithUnitType.php

```diff
@@ -51,6 +52,17 @@ public function configureOptions(OptionsResolver $resolver)
             'widget' => 'single_text',
             'unit' => 'unit',
         ]);
+
+        $resolver->setNormalizer('attr', function (Options $options, $value) {
```

Comment:
I think you could use a FormExtension for this that would be used for both NumberType and TextWithInputType so you wouldn't need to modify the form theme anymore And it would still work regardless of which form theme is used